src/argus_htmx/themes/utils.py

```diff
@@ -1,21 +1,47 @@
+from importlib.resources import files
+from pathlib import Path
+from re import findall
+
 from django.conf import settings
-from argus_htmx import settings as argus_htmx_settings
+from django.core.exceptions import ImproperlyConfigured
 
+from argus_htmx import settings as default_htmx_settings
 
-def get_themes():
-    return getattr(settings, "DAISYUI_THEMES", argus_htmx_settings.DAISYUI_THEMES)
 
+__all__ = ["get_theme_names"]
 
-def get_theme_names():
-    themes = get_themes()
+
+def get_themes_from_setting():
+    themes_setting = getattr(settings, "DAISYUI_THEMES", default_htmx_settings.DAISYUI_THEMES)
     theme_names = []
-    for theme in themes:
+    for theme in themes_setting:
         if isinstance(theme, str):
             theme_names.append(theme)
         elif isinstance(theme, dict):
             theme_names.extend(theme.keys())
     return theme_names
 
 
+def get_themes_from_css():
+    THEME_NAME_RE = "(?P<theme>\w+)"
+    DATA_THEME_RE = f"\[data-theme={THEME_NAME_RE}\]"
+
+    static_url = Path(settings.STATIC_URL).relative_to("/")
+    stylesheet_path = static_url / default_htmx_settings.STYLESHEET_PATH
```

Comment:
this needs to look at the django.conf.settings since it might  be overridden in another settings file (we do this)